### PR TITLE
frontend: Add a SDL2 headless window

### PR DESCRIPTION
**Pull request by Thog**
This is an all integrated experience that only depends on SDL2.
Every settings are configurable via commandline.

User can control input mapping via presaved profile from GTK3 or use the default preset.

It still need to be decided if we want to put sidebar information in the title bar.

image

### CODE REVIEW

File: Ryujinx.Headless.SDL2/OpenGL/OpenGLWindow.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+                GL.LoadBindings(new OpenToolkitBindingsContext());
+
+                SDL_GL_SetAttribute(SDL_GLattr.SDL_GL_SHARE_WITH_CURRENT_CONTEXT, 0);
+
+                SDL_GL_MakeCurrent(windowHandle, IntPtr.Zero);
+
+                return new SDL2OpenGLContext(context, windowHandle);
+            }
+
```
Comment (AcK77) [inline Ryujinx.Headless.SDL2/OpenGL/OpenGLWindow.cs:72]: Remove extra empty line

File: Ryujinx.Headless.SDL2/OpenGL/OpenGLWindow.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+            public void Dispose()
```
Comment (AcK77) [inline Ryujinx.Headless.SDL2/OpenGL/OpenGLWindow.cs:74]: Dispose are usually at the end of the class

File: Ryujinx.Headless.SDL2/Program.cs [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+                {
+                    Logger.Info?.Print(LogClass.Application, $"{index} not configured, defaulting to default keyboard.");
+
+                    // Default to keyboard
+                    inputId = "0";
+                }
+                else
+                {
+                    Logger.Info?.Print(LogClass.Application, $"{index} not configured");
+                    return null;
```
Comment (AcK77) [inline Ryujinx.Headless.SDL2/Program.cs:92]: Add a new empty line before

File: Ryujinx.Headless.SDL2/Program.cs [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+                    gamepad.Dispose();
+                }
+
+                return;
+            }
+
+            if (option.InputPath == null)
+            {
+                Logger.Error?.Print(LogClass.Application, "Please provide a file to load");
+                return;
```
Comment (AcK77) [inline Ryujinx.Headless.SDL2/Program.cs:320]: Add a new empty line before

File: Ryujinx.Headless.SDL2/WindowBase.cs [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+
+            MainLoop();
+
+            renderLoopThread.Join();
+            nvStutterWorkaround.Join();
+
+            Exit();
+        }
+
+        public void Dispose()
```
Comment (AcK77) [inline Ryujinx.Headless.SDL2/WindowBase.cs:361]: Dispose are usually at the end of the class (same for the next method)

**Review by AcK77 — CHANGES_REQUESTED** [review]

**Review by AcK77 — APPROVED** [review]

**Comment by albfan** [comment]
As I'm trying to use to integrate with Kodi https://github.com/xbmc/repo-plugins/pull/3510

A signal would be great to trigger "stop emulation"

**Comment by albfan** [comment]
Nvm this is merged, continues on #2131

**Pull request merged**